### PR TITLE
Another round of atom mapping fixes

### PR DESCRIPTION
**Pull request by kfir4444**
Finally, no more reactions in work in progress

### CODE REVIEW

File: arc/mapping/driver.py [hunk 1]
```diff
@@ -248,14 +248,16 @@ def map_rxn(rxn: 'ARCReaction',
...
+    except (ValueError, IndexError, ActionError, AtomTypeError) as e:
+        logger.warning(e)
 
     r_cuts, p_cuts = update_xyz(r_cuts), update_xyz(p_cuts)
 
     #step 4:
     pairs_of_reactant_and_products = pairing_reactants_and_products_for_mapping(r_cuts, p_cuts)
-
+    if len(p_cuts):
+        logger.error(f"Could not find isomorphism for scissored species: {[cut.mol.smiles for cut in p_cuts]}")
```
Comment (alongd) [inline arc/mapping/driver.py:259]: I don't think an `ARCSpecies` has a `.smiles` attribute

File: arc/mapping/driver.py [hunk 2]
```diff
@@ -248,14 +248,16 @@ def map_rxn(rxn: 'ARCReaction',
...
-        return None
+    except (ValueError, IndexError, ActionError, AtomTypeError) as e:
+        logger.warning(e)
 
     r_cuts, p_cuts = update_xyz(r_cuts), update_xyz(p_cuts)
 
     #step 4:
     pairs_of_reactant_and_products = pairing_reactants_and_products_for_mapping(r_cuts, p_cuts)
-
+    if len(p_cuts):
```
Comment (alongd) [inline arc/mapping/driver.py:258]: should this be the opposite, `if not len(p_cuts)`?

File: arc/reaction_test.py [hunk 3]
```diff
@@ -1325,13 +1326,6 @@ def test_get_atom_map(self):
         self.assertIn(tuple(rxn.atom_map[5:7]+rxn.atom_map[8:]), permutations([2, 3, 4, 5]))
         self.assertTrue(check_atom_map(rxn))
 
-    @work_in_progress
```
Comment (alongd) [inline arc/reaction_test.py:1328]: hooray!

File: arc/mapping/driver.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
                                 RESERVED_FINGERPRINT_KEYS,)
 from arc.common import logger
 
-from rmgpy.exceptions import ActionError
+from rmgpy.exceptions import ActionError, AtomTypeError
```
Comment (alongd) [inline arc/mapping/driver.py:32]: squash with the "Improved logging of atom mapping" commit?

File: arc/mapping/engine.py [hunk 5]
```diff
@@ -1122,13 +1122,15 @@ def make_bond_changes(rxn: 'ARCReaction',
                             atom1 = atom
                         elif int(atom.label) == indicies[1]:
                             atom2 = atom
-                    if atom1.radical_electrons == 0:
+                    if atom1.radical_electrons == 0 and atom2.radical_electrons == 0: # Both atoms doesn't have any radicals, but their bond is changing. There probably is resonance, so this will not effect the isomorphism check.
```
Comment (alongd) [inline arc/mapping/engine.py:1125]: Very minor typos: `don't` instead of `doesn't`, `affect` instead of `effect`

**Review by alongd — COMMENTED** [review]
Thanks, looking good!
I guess the last commit addresses the issue we saw with partial charges?

File: arc/mapping/driver.py (hunk 2 above)
Comment (kfir4444) [reply arc/mapping/driver.py:258]: No, after `pairs_of_reactant_and_products` `p_cuts` should be an empty list

File: arc/mapping/driver.py (hunk 1 above)
Comment (kfir4444) [reply arc/mapping/driver.py:259]: Do you mean in: `[cut.mol.smiles for cut in p_cuts]`? because the `.mol` object does have `.smiles` property.

**Review by alongd — APPROVED** [review]
Thanks!

**Pull request merged**